x-pack/plugin/ml/build.gradle

```diff
@@ -22,7 +22,7 @@ repositories {
           artifact()
         }
         patternLayout {
-          artifact "maven/org/elasticsearch/ml/ml-cpp/[revision]/[module]-[revision].[ext]"
+          artifact "maven/org/elasticsearch/ml/ml-cpp/[revision]/[module](-[classifier])-[revision].[ext]"
```

Comment:
With the current changes in https://github.com/elastic/ml-cpp/pull/2224 the classifier will need to go after the revision, not before.  I know that's different to what's currently uploaded to S3, so we'll just have to make sure the C++ PR is merged first and new artifacts uploaded to S3 before this PR is merged.  For testing I can also duplicate the current artifacts on S3 with `deps`/`nodeps` moved to the end so that builds of this PR can succeed.

Reply:
The manually duplicated files are in place so that this PR can test the download case if we continue down the road of using the classifier for `deps`/`nodeps`.

<img width="1269" alt="Screenshot 2022-03-03 at 11 53 08" src="https://user-images.githubusercontent.com/7405510/156559971-b572324c-4f12-4f06-a3df-92be504130c9.png">